ci/redmine_bugzilla.py

```diff
@@ -0,0 +1,153 @@
+    :param password: The password to connect with
+    :type password: basestring
+
+    :return: An instantiated Bugzilla connection object.
+    """
+    return RHBugzilla(url='%s/xmlrpc.cgi' % BUGZILLA_URL, user=user, password=password)
+
+
+def get_redmine_connection(key):
+    """
+    Return the Redmine connection.
+
+    :param key: The api key to connect with
+    :type key: basestring
+
+    :return: An instantiated Redmine connection object.
+    """
+    return Redmine(REDMINE_URL, key=key)
+
+
+def main():
+    config = ConfigParser.ConfigParser()
+    config.read(os.environ['REDMINE_BUGZILLA_CONF'])
+
+    username = config.get('bugzilla', 'username')
+    password = config.get('bugzilla', 'password')
+    key = config.get('redmine', 'key')
+
+    redmine = get_redmine_connection(key)
+    BZ = get_bugzilla_connection(username, password)
+
+    redmine_issues = [issue for issue in redmine.issue.filter(query_id=24)]
+
+    non_closed_bug_with_ext_tracker = BUGZILLA_URL + '/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&bug_status=POST&bug_status=MODIFIED&bug_status=ON_DEV&bug_status=ON_QA&bug_status=VERIFIED&bug_status=RELEASE_PENDING&columnlist=priority%2Cbug_severity%2Cbug_status%2Cshort_desc%2Cchangeddate%2Ccomponent%2Ctarget_release%2Cassigned_to%2Creporter&f1=external_bugzilla.url&list_id=3309842&o1=substring&query_format=advanced&v1=pulp.plan.io'
+    bugzilla_bugs = BZ.query(RHBugzilla.url_to_query(non_closed_bug_with_ext_tracker))
+
+    links_issues_record = ''
+    ext_bug_record = ''
+
+    for issue in redmine_issues:
+        for custom_field in issue.custom_fields.resources:
+            if custom_field['name'] == 'Bugzilla':
+                bug = BZ.getbug(int(custom_field['value']))
+                links_back = False
+                for external_bug in bug.external_bugs:
+                    if external_bug['type']['description'] == 'Pulp Redmine' and external_bug['ext_bz_bug_id'] == str(issue.id):
+                        ext_params = {}
+                        if external_bug['ext_description'] != issue.subject:
+                            ext_params['ext_description'] = issue.subject
+                        if external_bug['ext_status'] != issue.status.name:
+                            ext_params['ext_status'] = issue.status.name
+                        if external_bug['ext_priority'] != issue.priority.name:
+                            ext_params['ext_priority'] = issue.priority.name
+                        if len(ext_params.keys()) > 0:
+                            ext_bug_record += 'Bugzilla bug %s updated from upstream bug %s with %s\n' % (bug.id, issue.id, ext_params)
+                            ext_params['ids'] = external_bug['id']
+                            BZ.update_external_tracker(**ext_params)
+
+                        # Remove the bug list gotten via search so we don't examine it again when bugzilla_bugs are iterated through
+                        bug_in_search_index_to_remove = -1
```

Comment:
Good suggestion. I originally used None, but when I checked the value I used `if bug_in_search_index_to_remove` which caused a bug where it wouldn't go in when `bug_in_search_index_to_remove` is 0. I opted for assigning it to `None` and evaluating using `foo is not None`.